lms/static/scripts/frontend_apps/components/test/GradingControls-test.js

```diff
@@ -271,6 +263,49 @@ describe('GradingControls', () => {
     );
   });
 
+  context('when has unsaved grade changes', () => {
+    const setUnsavedChanges = wrapper =>
+      act(() => {
+        wrapper.find('SubmitGradeForm').props().onUnsavedChanges(true);
+      });
+    const getSelectedStudent = wrapper =>
+      wrapper.find('StudentSelector').props().selectedStudent;
+
+    [
+      // Selected student should still be the first
+      {
+        discardChanges: false,
+        expectedStudentAfterChange: () => fakeStudents[0],
+      },
+
+      // Selected student should change from first to second
+      {
+        discardChanges: true,
+        expectedStudentAfterChange: () => fakeStudents[1],
+      },
+    ].forEach(({ discardChanges, expectedStudentAfterChange }) => {
+      it('"cancels" selected user change if instructor selected to continue editing', async () => {
+        const wrapper = renderGrader();
+        await selectFirstStudent(wrapper);
+
+        await setUnsavedChanges(wrapper);
```

Comment:
The `act` callback in `setUnsavedChanges` is synchronous. I believe this works because calling `await` here pauses this code for one microtask, allowing other async activity triggered by `onUnsavedChanges` to complete. This would break if the `onUnsavedChanges` prop callback changed and took more than one microtask to execute.

If the code was like this:

```js
const setUnsavedChanges = wrapper => act(async () => {
  await wrapper.find('SubmitGradeForm').props().onUnsavedChanges(true);
});
```

Then it would mean that `await setUnsavedChanges` would wait until the async `onUnsavedChanges` prop finished, regardless of how long it took.

Reply:
Ok, I think I get it. I have applied a slightly different variation of the above, which should be equivalent:

```ts
const setUnsavedChanges = wrapper =>
  act(() => wrapper.find('SubmitGradeForm').props().onUnsavedChanges(true));
```

I did the same with the "global" `selectFirstStudent`, and fixed an older test in the process, which was not waiting for it.

Reply:
Yes, this should work the same.